src/test/scala/org/broadinstitute/dsde/workbench/sam/service/UserServiceSpecs/UpdateUserSpec.scala

```diff
@@ -79,5 +82,146 @@ class UpdateUserSpec extends UserServiceTestTraits {
+
+      // Assert
+      assert(response.nonEmpty, "Updating a user with an empty azureB2CId should return an object")
+      assert(
+        response.get equals userWithBothIds.copy(azureB2CId = None),
+        "Updating a user with an empty azureB2CId should return the corresponding user after update."
+      )
+    }
+  }
+
+  describe("an existing user's azureB2CId, who has a null googleSubjectId") {
+    describe("with an null azureB2CId") {
+      it("should throw an exception") {
+        // Arrange
+        val azureUser = genWorkbenchUserAzure.sample.get
+        val directoryDAO = new MockDirectoryDAO
+        val cloudExtensions = StatefulMockCloudExtensionsBuilder(directoryDAO).build
+        val tosService = mock[TosService]
+        val userService = new UserService(directoryDAO, cloudExtensions, Seq.empty, tosService)
+
+        val updateRequest = AdminUpdateUserRequest(Some(AzureB2CId("null")), None, None, None)
+
+        TestSupport.runAndWait(directoryDAO.createUser(azureUser, samRequestContext))
+
+        // Act
+        val error = intercept[WorkbenchExceptionWithErrorReport] {
+          runAndWait(userService.updateUserCrud(azureUser.id, updateRequest, samRequestContext))
+        }
+
+        // Assert
+        assert(
+          error.errorReport.statusCode.value equals StatusCodes.BadRequest,
+          "Nulling a user's azureB2CId when the user has no googleSubjectId should have a Bad Request status"
+        )
+        assert(
+          error.errorReport.message equals "Cannot null a user's azureB2CId when the user has no googleSubjectId",
+          "Nulling a user's azureB2CId when the user has no googleSubjectId should send back a invalid update message"
+        )
+      }
+    }
+  }
+
+  describe("an existing user's googlesSubjectId, who has a null azureB2CId") {
+    describe("with an null googleSubjectId") {
+      it("should throw an exception") {
+        // Arrange
+        val googleUser = genWorkbenchUserGoogle.sample.get
+        val directoryDAO = new MockDirectoryDAO
+        val cloudExtensions = StatefulMockCloudExtensionsBuilder(directoryDAO).build
+        val tosService = mock[TosService]
+        val userService = new UserService(directoryDAO, cloudExtensions, Seq.empty, tosService)
+
+        val updateRequest = AdminUpdateUserRequest(None, Some(GoogleSubjectId("null")), None, None)
+
+        TestSupport.runAndWait(directoryDAO.createUser(googleUser, samRequestContext))
+
+        // Act
+        val error = intercept[WorkbenchExceptionWithErrorReport] {
+          runAndWait(userService.updateUserCrud(googleUser.id, updateRequest, samRequestContext))
+        }
+
+        // Assert
+        assert(
+          error.errorReport.statusCode.value equals StatusCodes.BadRequest,
+          "Nulling a user's azureB2CId when the user has no googleSubjectId should have a Bad Request status"
```

Comment:
Should this be flipped? (same below):
```suggestion
          "Nulling a user's googleSubjectId when the user has no azureB2CId should have a Bad Request status"
```